data-collector/src/main/kotlin/com/d3/datacollector/model/BillingModel.kt

```diff
@@ -65,4 +88,22 @@ data class Billing(
     protected fun onUpdate() {
         updated = Date().time
     }
+
+    override fun hashCode() =
+        (feeDescription + domainName + billingType + asset + destination +
+                minAmount.toPlainString() + maxAmount.toPlainString()).hashCode()
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
What about the null check at the beginning of the method? If 'other' is null, then we can return false at the beginning.